src/mame/sony/news_68k_iop.cpp

```diff
@@ -0,0 +1,1150 @@
+ *   - AST (Asynchronous System Trap) emulation
+ *   - System cache emulation
+ *   - Expansion slots (I/O Bus and VMEBus)
+ *   - Networking is very flaky, especially on NEWS-OS 4.
+ *   - Graphics, kbms, and parallel port emulation
+ *   - Hyperbus handshake for IOP and CPU accesses. The bus has arbitration circuitry to prevent bus contention when both the CPU and IOP are trying to access the hyperbus (RAM and VME)
+ */
+
+#include "emu.h"
+
+#include "news_hid.h"
+#include "news_020_mmu.h"
+#include "news_iop_scsi.h"
+
+#include "bus/nscsi/cd.h"
+#include "bus/nscsi/hd.h"
+#include "bus/nscsi/tape.h"
+#include "bus/rs232/rs232.h"
+
+#include "cpu/m68000/m68020.h"
+
+#include "formats/pc_dsk.h"
+
+#include "imagedev/floppy.h"
+
+#include "machine/am79c90.h"
+#include "machine/msm58321.h"
+#include "machine/ncr5380.h"
+#include "machine/nscsi_bus.h"
+#include "machine/pit8253.h"
+#include "machine/ram.h"
+#include "machine/timekpr.h"
+#include "machine/upd765.h"
+#include "machine/z80scc.h"
+#include "machine/bankdev.h"
+
+#define LOG_INTERRUPT (1U << 1)
+#define LOG_ALL_INTERRUPT (1U << 2)
+#define LOG_LED (1U << 3)
+#define LOG_MEMORY (1U << 4)
+#define LOG_PANEL (1U << 5)
+#define LOG_MEMORY_ERROR (1U << 6)
+#define LOG_TIMER (1U << 7)
+#define LOG_SCSI (1U << 8)
+#define LOG_AST (1U << 9)
+#define LOG_RTC (1U << 10)
+
+#define VERBOSE (LOG_GENERAL)
+
+#include "logmacro.h"
+
+namespace
+{
+    using namespace std::literals::string_view_literals;
+
+    class news_iop_state : public driver_device
+    {
+    public:
+        static constexpr feature_type unemulated_features() { return feature::GRAPHICS; }
+
+        news_iop_state(machine_config const &mconfig, device_type type, char const *tag)
+            : driver_device(mconfig, type, tag),
+              m_iop(*this, "iop"),
+              m_cpu(*this, "cpu"),
+              m_mmu(*this, "mmu"),
+              m_ram(*this, "ram"),
+              m_eprom(*this, "eprom"),
+              m_idrom(*this, "idrom"),
+              m_rtc(*this, "rtc"),
+              m_interval_timer(*this, "interval_timer"),
+              m_scc_external(*this, "scc_external"),
+              m_scc_peripheral(*this, "scc_peripheral"),
+              m_net(*this, "net"),
+              m_fdc(*this, "fdc"),
+              m_scsi(*this, "scsi:7:am5380"),
+              m_scsi_dma(*this, "scsi_dma"),
+              m_dip_switch(*this, "FRONT_PANEL"),
+              m_serial(*this, "serial%u", 0U)
+        {
+        }
+
+        void nws831(machine_config &config);
+        void init_common();
+
+    protected:
+        // driver_device overrides
+        virtual void machine_start() override;
+        virtual void machine_reset() override;
+
+        // address maps
+        void iop_map(address_map &map);
+        void iop_autovector_map(address_map &map);
+        void mmu_map(address_map &map);
+        void cpu_map(address_map &map);
+
+        // machine config
+        void common(machine_config &config);
```

Comment:
Please put `ATTR_COLD` on cold lifecycle functions (here and in devices, too).

Reply:
I added it in a bunch of places, if there is anything else that should have it let me know :)